core/controllers/suggestion.py

```diff
@@ -485,11 +485,19 @@ def _construct_exploration_suggestions(suggestions):
 
     suggestion_dicts = []
     for suggestion in suggestions:
-        content_html = exp_id_to_exp[suggestion.target_id].get_content_html(
-            suggestion.change.state_name, suggestion.change.content_id)
-        suggestion_dict = suggestion.to_dict()
-        suggestion_dict['exploration_content_html'] = content_html
-        suggestion_dicts.append(suggestion_dict)
+        available_states = exp_id_to_exp[suggestion.target_id].states
+        available_content_ids = []
+
+        if suggestion.change.state_name in available_states:
+            available_content_ids = available_states[
```

Comment:
Done

Reply:
I noticed a couple of issues:
- The name here suggests a list of content IDs, but that is not what this represents.
- You're only declaring available_content_ids in order to figure out whether the given content ID actually exists or not in the state. I think, rather than make _get_all_translatable_content() public and have it return a whole bunch of stuff here which this function doesn't really need, it would be better to just implement a state.has_content_id() method that returns a True/False boolean. In general it is a good design principle to minimize exposure of functions to data that that function doesn't need.